src/modules/roc_audio/pcm_funcs.cpp

```diff
@@ -43,9 +43,9 @@ size_t pcm_encode_samples(void* out_data,
                           size_t out_offset,
                           const sample_t* in_samples,
                           size_t in_n_samples,
-                          packet::channel_mask_t in_chan_mask) {
-    const packet::channel_mask_t out_chan_mask = packet::channel_mask_t(1 << NumCh) - 1;
-    const packet::channel_mask_t inout_chan_mask = in_chan_mask | out_chan_mask;
+                          SampleSpec& sample_spec) {
+    const channel_mask_t out_chan_mask = channel_mask_t(1 << NumCh) - 1;
+    const channel_mask_t inout_chan_mask = sample_spec.getChannels() | out_chan_mask;
```

Comment:
* Let's rename sample_spec to in_sample_spec to make it clear is it for input or output.
* Let's cache sample_spec.getChannels() in a variable, we can call it in_chan_mask.

Same for pcm_decode_samples, but with "out" instead of "in".